feat(search) Improve support for boolean filters

markstory opened the pull request:
Previously you would have to use '1' and '0' to interact with boolean values. This was inconsistent with how these values are shown in the UI.

By accepting yes/no/true/false we can work with values more commonly associated with booleans. This is preparatory work for improving the search interface for `error.handled` and adding `error.unhandled` as a field. I'm also going to explore improving the ergonomics around `device.charging` and `device.online` as they are boolean-ish fields as well.

markstory committed:
feat(search) Improve support for boolean filters

Previously you would have to use '1' and '0' to interact with boolean
values. This was inconsistent with how these values are shown in the UI.

By accepting yes/no/true/false we can work with values more commonly
associated with booleans.

wmak commented on src/sentry/api/event_search.py:
```diff
@@ -394,6 +397,26 @@ def visit_paren_term(self, node, children):
+            return self.visit_numeric_filter(node, (search_key, sep, "=", search_value))
+
+        if search_key.name in self.boolean_keys:
+            if search_value.text in ("yes", "true", "1"):
```
Could we add a `.lower()` so that we accept `YES` and `TrUe` as well?

evanh replied:
That should be added to the regex above. Otherwise `error.handled:Yes` won't be interpreted as a boolean filter at all.

markstory replied:
Made it case insensitive now.

markstory committed:
Make boolean values case insensitive.

markstory committed:
Remove yes/no

Having two forms of boolean expressions is wasteful. true/false is
simple and aligns with how developers use booleans.

markstory committed:
Fix test.

markstory committed:
Fix more tests.

markstory committed:
Merge branch 'master' into feat/boolean-values

adhiraj commented:
heck yes. can we also stop showing 0, 1, and None in the autocomplete?
